src/components/header/connect.tsx

```diff
@@ -1,34 +1,80 @@
 'use client'
 
-import { ConnectButton } from '@rainbow-me/rainbowkit'
-import { useEffect } from 'react'
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import { Button } from '@headlessui/react'
+import { CoreHelperUtil } from '@web3modal/core'
+import { useWeb3Modal } from '@web3modal/wagmi/react'
+import { useMemo } from 'react'
+import { formatUnits } from 'viem'
+import { useAccount, useAccountEffect, useBalance } from 'wagmi'
 
+import { NetworkSelect } from '@/components/header/network-select'
 import { useAnalytics } from '@/lib/hooks/use-analytics'
-import { useNetwork } from '@/lib/hooks/use-network'
-import { useWallet } from '@/lib/hooks/use-wallet'
+import { shortenAddress } from '@/lib/utils'
+import { emojiAvatarForAddress } from '@/lib/utils/emojiAddressAvatar'
 
 export const Connect = () => {
+  const { address, isConnected } = useAccount()
+  const { open } = useWeb3Modal()
   const { logConnectWallet } = useAnalytics()
-  const { chainId } = useNetwork()
-  const { address } = useWallet()
 
-  useEffect(() => {
-    if (address === undefined || chainId === undefined) return
-    logConnectWallet(address, chainId)
-  }, [address, chainId, logConnectWallet])
+  const { data: balance } = useBalance({
+    query: {
+      initialData: {
+        value: BigInt(0),
+        decimals: 18,
+        symbol: 'ETH',
+        formatted: '0',
+      },
+    },
+    address,
+  })
+
+  useAccountEffect({
+    onConnect: ({ address, chainId }) => {
+      logConnectWallet(address, chainId)
+    },
+  })
+
+  const avatar = useMemo(() => {
+    if (!address) return null
+
+    return emojiAvatarForAddress(address)
+  }, [address])
 
   return (
-    <ConnectButton
-      label='Connect'
-      showBalance={{
-        smallScreen: false,
-        largeScreen: true,
-      }}
-      accountStatus={{
-        smallScreen: 'avatar',
-        largeScreen: 'full',
-      }}
-      chainStatus={{ smallScreen: 'icon', largeScreen: 'full' }}
-    />
+    <div className='text-ic-white flex gap-4 font-bold'>
+      {isConnected && address ? (
+        <>
+          <NetworkSelect />
+          <Button
+            className='bg-ic-black flex items-center gap-2 rounded-md px-4 py-2 transition-all duration-300 hover:scale-[1.04]'
+            onClick={() => open({ view: 'Account' })}
+          >
+            <p className='hidden md:block'>
+              {CoreHelperUtil.formatBalance(
+                formatUnits(balance!.value, balance!.decimals),
+                balance!.symbol,
```

Comment:
for these please use our [utils](https://github.com/IndexCoop/index-app/blob/master/src/lib/utils/index.ts) to stay more lib agnostic in code.

Reply:
Well I would have, but this thing is formatting significant decimals, which is a nice approach. Would you like me to get the code, or just use this thing inside the utils?